lib/macho/exceptions.rb

```diff
@@ -3,6 +3,10 @@ module MachO
   class MachOError < RuntimeError
   end
 
+  # A generic Mach-O error during file modification.
+  class MachOModificationError < MachOError
```

Comment:
I'm a little confused about how we're using "fatal" here, actually: do we mean "fatal" as in "halts the program" or as in "would do something very bad [to the Mach-O] if it were allowed to continue"?